src/coreclr/jit/importer.cpp

```diff
@@ -4235,6 +4235,24 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
             break;
         }
 
+        case CORINFO_FIELD_STATIC_DATASEGMENT:
+        {
+#ifdef FEATURE_READYTORUN
+            assert(opts.IsReadyToRun());
```

Comment:
```suggestion
            assert(opts.IsReadyToRun());
```
I would delete this assert. Nothing in the Jit depends on it for the correct implementation of this kind of static field.